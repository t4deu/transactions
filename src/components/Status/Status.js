// @flow
// external imports
import React from 'react';
import { View, Text } from 'react-native';
import { Card } from 'react-native-material-ui';
// types
import type { Status as CurrentStatus } from '../../state/Ducks/Transactions/Types';
// containers & components
import { toCurrency } from '../../Utils';
// styles
import styles from './Status.styles';

type Props = {
  currentStatus: CurrentStatus,
};

const Status = (props: Props) => {
  const {
    total, creditTotal, debitTotal, isSaver,
  } = props.currentStatus;

  return (
    <View style={styles.painel}>
      <Card>
        <View style={styles.card}>
          <Text style={styles.text}>
            Você economizou <Text style={styles.currency}>{toCurrency(creditTotal)}</Text> e gastou{' '}
            <Text style={styles.currency}>{toCurrency(debitTotal)}</Text>. Atualmente você tem:
          </Text>
          <Text style={styles.total}>{toCurrency(total)}</Text>
          {isSaver && <Text style={styles.advice}>Parabéns! Você sabe usar deu dinheiro</Text>}
          {!isSaver && (
            <Text style={styles.advice}>Cuidado! Tá bom de começar a dar uma maneirada</Text>
          )}
        </View>
      </Card>
    </View>
  );
};

export default Status;
