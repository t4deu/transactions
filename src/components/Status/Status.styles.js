export default {
  painel: {
    flex: 1,
    paddingVertical: 20,
    height: 200,
    backgroundColor: '#fafafa',
  },
  card: {
    padding: 20,
  },
  text: {
    fontSize: 16,
  },
  currency: {
    color: '#8600c9',
    fontWeight: 'bold',
  },
  total: {
    color: '#8600c9',
    fontWeight: 'bold',
    fontSize: 18,
    textAlign: 'center',
    marginVertical: 5,
  },
  advice: {
    fontStyle: 'italic',
  }
};
