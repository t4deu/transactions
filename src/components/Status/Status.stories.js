import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { View } from 'react-native';

import Status from './Status';

storiesOf('Status', module)
  .addDecorator(story => <View style={{ flex: 1, marginTop: 200 }}>{story()}</View>)
  .add('Saver', () => (
    <Status
      currentStatus={{
        debitTotal: -200,
        creditTotal: 4000,
        total: 3800,
        isSaver: true,
      }}
    />
  ))
  .add('Spender', () => (
    <Status
      currentStatus={{
        debitTotal: -1500,
        creditTotal: 200,
        total: -1300,
        isSaver: false,
      }}
    />
  ));
