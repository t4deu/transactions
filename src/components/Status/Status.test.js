import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Status from './Status';

describe('<Status>', () => {
  it('renders without issues for saver', () => {
    const component = shallow(<Status
      currentStatus={{
          debitTotal: -200,
          creditTotal: 4000,
          total: 3800,
          isSaver: true,
        }}
    />);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders without issues for spender', () => {
    const component = shallow(<Status
      currentStatus={{
          debitTotal: 200,
          creditTotal: -4000,
          total: -3800,
          isSaver: false,
        }}
    />);

    expect(toJson(component)).toMatchSnapshot();
  });
});
