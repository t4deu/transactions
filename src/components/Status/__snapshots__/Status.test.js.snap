// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Status> renders without issues for saver 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fafafa",
      "flex": 1,
      "height": 200,
      "paddingVertical": 20,
    }
  }
>
  <Card
    onPress={null}
    style={Object {}}
  >
    <View
      style={
        Object {
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 16,
          }
        }
      >
        Você economizou 
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#8600c9",
              "fontWeight": "bold",
            }
          }
        >
          R$ 4,000,00
        </Text>
         e gastou
         
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#8600c9",
              "fontWeight": "bold",
            }
          }
        >
          R$ -200,00
        </Text>
        , atualmente você tem:
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#8600c9",
            "fontSize": 18,
            "fontWeight": "bold",
            "marginVertical": 5,
            "textAlign": "center",
          }
        }
      >
        R$ 3,800,00
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontStyle": "italic",
          }
        }
      >
        Parabéns! Você sabe usar deu dinheiro
      </Text>
    </View>
  </Card>
</View>
`;

exports[`<Status> renders without issues for spender 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fafafa",
      "flex": 1,
      "height": 200,
      "paddingVertical": 20,
    }
  }
>
  <Card
    onPress={null}
    style={Object {}}
  >
    <View
      style={
        Object {
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 16,
          }
        }
      >
        Você economizou 
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#8600c9",
              "fontWeight": "bold",
            }
          }
        >
          R$ -4,000,00
        </Text>
         e gastou
         
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#8600c9",
              "fontWeight": "bold",
            }
          }
        >
          R$ 200,00
        </Text>
        , atualmente você tem:
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#8600c9",
            "fontSize": 18,
            "fontWeight": "bold",
            "marginVertical": 5,
            "textAlign": "center",
          }
        }
      >
        R$ -3,800,00
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontStyle": "italic",
          }
        }
      >
        Cuidado! Tá bom de começar a dar uma maneirada
      </Text>
    </View>
  </Card>
</View>
`;
